boards/FSM/Src/App/App_FsmWorld.c

```diff
@@ -121,3 +133,20 @@ struct RgbLedSequence *
 {
     return world->rgb_led_sequence;
 }
+
+void App_FsmWorld_RegisterSignal(struct FsmWorld *world, struct Signal *signal)
+{
+    assert(world->num_signals < MAX_NUM_SIGNALS);
+    world->signals[world->num_signals] = signal;
+    world->num_signals++;
+}
+
+void App_FsmWorld_UpdateSignals(
```

Comment:
Not quite sure where this function should be called. If it's put inside `RunTask1kHz`, then the signals won't be updated in `Test_StateMachine` so I thought it'd be more appropriate to call it in a state machine. If I go with that approach, I run into  two problems:

- I will have to update the state machine library to support 1kHz because it only supports 1Hz and 100Hz at the moment (Not really a "problem", but just something we need to add)
- Inside the state machine, there is no "time" so I'm not sure what value to pass in for `current_time_ms`. A potential solution would be to add `App_FsmWorld_GetTimeMs` by passing in `get_time_ms` to `App_FsmWorld_Create()`

Reply:
I think we need to call it separately. It might be worth having a test util function called `letTimePass(unsigned int time_ms, StateMachine* state_machine)` or something that manually calls both this and the 1hz and 1khz functions? (hopefully that makes sense, I can write out more of an example if not)

Reply:
Can you write an example how `UpdateSignal` would get used in embedded and test targets?

Reply:
```
void letTimePass(struct SharedStateMachine* state_machine, unsigned int time_ms){
    for (unsigned int i = 0; i < time_ms; i++){
        if (i % 1000 == 0) {
            app_SharedStateMachine_tick1Hz(state_machine);
        }
        app_SharedStateMachine_tick1kHz(state_machine);
    }
    // So we'd probably have to do something where this is a virtual function
    // that is implemented in each of the board-specific test fixtures. I've
    // put an example down below for erferece
    updateSignals(state_machine);
}

// This would have to be board-specific. We can enforce that people implement
// it if we have a shared test fixture that all the board-specific test 
// fixtures extend
void updateSignals(struct SharedStateMachine* state_machine){
    // This specific example is for the DCM
    struct DcmWorld* world = app_SharedStateMachine_getWorld(state_machine);
    app_DcmWorld_updateSignals(world);
}
```

Reply:
This is more or less what I had in mind, looks good!

Reply:
Also slightly off topic, but later we would probably call `SetCurrentTimeInMilliseconds` from #644 in `letTimePass` as well right?